### PR TITLE
Fixes star not highlighted issue

### LINKED ISSUE #1298: Starring the Circuit has problems
**Describe the bug**
On opening any of the favourite circuits, the icon displays it the opposite, although the working is completely fine.

**To Reproduce**
1. Go to your favourite projects
2. Open one of the circuits and observe the icon should be filled.

**Expected behavior**
The icon should appear filled if the circuit has been already starred.

**Screenshots**
Untitled_ Mar 25, 2020 6_27 PM

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Nitish145**
Fixes #1298

#### Describe the changes you have made in this PR -
Selectively adds **fa-star-o** or **fa-star** based on _is_star_for_project_and_user_

### Screenshots of the changes (If any) -

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Nitish145** [comment]
@nitin10s @tachyons please check

File: app/views/projects/show.html.erb [hunk 1]
```diff
@@ -81,7 +81,7 @@
               <hr>
               <a href="<%= simulator_edit_path(@project) %>" class="card-link btn btn-primary btn-post fa-post" target="_blank">Launch simulator</a>
               <% if user_signed_in? %>
-                <i id="toggleStarButton" class="fa fa-star-o btn btn-primary fa-post noSelect" aria-hidden="true" style="font-size:14px;margin:0px;" onclick="toggleStar()"></i>
+                <i id="toggleStarButton" class="fa <%= Star.find_by(user_id: current_user.id, project_id: @project.id).nil? ? "fa-star-o" : "fa-star" %> btn btn-primary fa-post noSelect" aria-hidden="true" style="font-size:14px;margin:0px;" onclick="toggleStar()"></i>
```
Comment (tachyons) [inline app/views/projects/show.html.erb]: Move query to controller

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -128,7 +128,7 @@
         <% end %>
         <% if !policy(@project).user_access? %>
           <% if user_signed_in? %>
-            <i id="toggleStarButton" class="fa fa-star-o btn btn-primary fa-post btn-post noSelect" aria-hidden="true" style="font-size:13px;margin:0px;" style=<%= Star.find_by(user_id: current_user.id, project_id: @project.id).nil? ? "color:white" : "color:orange" %> onclick="toggleStar()"></i>
+            <i id="toggleStarButton" class="fa <%= Star.find_by(user_id: current_user.id, project_id: @project.id).nil? ? "fa-star-o" : "fa-star" %> btn btn-primary fa-post noSelect" aria-hidden="true" style="font-size:14px;margin:0px;" onclick="toggleStar()"></i>
```
Comment (tachyons) [inline app/views/projects/show.html.erb]: Move db queries to controller

Comment (Nitish145) [reply app/views/projects/show.html.erb]: Will do @tachyons

Comment (Nitish145) [reply app/views/projects/show.html.erb]: @tachyons please check !!

**Comment by Nitish145** [comment]
@tachyons the `id` is called on `nil (current_user)` because it is not defined in tests probably.. How can i tackle this to resolve failing build ?

**Comment by Nitish145** [comment]
@tachyons please check !!

**Comment by aayush-05** [comment]
LGTM, @tachyons please review this

**Comment by nitin10s** [comment]
@tachyons

**Comment by aayush-05** [comment]
Can we avoid merging this one?. Already fixed it here >> https://github.com/CircuitVerse/CircuitVerse/pull/1495 (i am just reducing the merge conflicts for me :p)

**Comment by aayush-05** [comment]
@tachyons @nitin10s this can be closed >> fixed in new UI. Thanks for the effort @Nitish145

**Pull request closed without merging**